python/check_address.py

```diff
@@ -52,14 +53,42 @@ def handle_msg(self, msg_pmt):
         else:
             address = packet[7:14]
 
+        if len(packet) > 20:
```

Comment:
What happens if `len(packet) <= 20`? It looks like `hbit` and `digi` are not defined, but they are read afterwards regardless.

Reply:
What happens if `len(packet) <= 20`?
->`hbit` and `digi` won't be defined.

Is it ok for you to initialize `hbit` and `digi` right before `len(packet) <= 20` by:
```
        hbit = 0
        digi = ""
```

Reply:
I guess not. What's the code supposed to do in this case? If repeater address checking has been enabled but the message doesn't even contain repeater addresses because it is too short, shouldn't it go directly to the `fail` output?

Reply:
If `len(packet) <=20`, then hbit is 0 and there is no digipeater (digi = "").
In this case, the regular sender / destination callsign check should and will be processed anyway.

Reply:
I had to read again the condition you wrote on lines 87-89 to understand what this feature is supposed to do. Let me rephrase: this feature will pass the frame to `ok` if either its destination address matches `address` or if the frame contains 3 addresses, the H bit is set (indicating that the frames has been repeated) and the repeater address (3rd address) matches `digicallsign`.

In that case I agree that just setting `hbit = 0` if `len(packet) <= 20` might be enough.

Reply:
> (...) if either its destination address matches `address` (...)
yes, destination or source (according to the chosen _direction_ "to" or "from")

Correct, that's the way I wanted it to work.